csrc/fusion_segmenter.cpp

```diff
@@ -3968,7 +4033,7 @@ void SegmentCandidateFinder::trySetUpMerge(
   }
 
   auto candidate_it = candidates.begin();
-  while (candidate_it != candidates.end() &&
+  while (candidate_it != candidates.end() && !candidate_it->group->isMerged() &&
```

Comment:
Added `isMerged`. This may not be necessary but just in case.

Reply:
IIUC, we are looking for non merged candidate?

Shouldn't the logic be the opposite? `candidate_it->group->isMerged()` instead?

Reply:
Thanks. Actually, this should be 
```suggestion
  while (candidate_it != candidates.end() && 
      (candidate_it->group->isMerged() ||
       !codeGenSupportedMerge(group, candidate_it->group))) {
       candidate_it++;
    }
```

Reply:
This while loop doesn't seem easy to understand. Will update.